### DISCUSSION TITLE
SKLearn Cross Validation And Qiskit ML

### DISCUSSION THREAD

**Discussion by Raubkatz** [Qiskit/.github#31] [Ideas]
Hi!

I'm currently working on some ML projects and wanted to try some new algorithms and ideas. My testing framework is based on the basic Cross-Validation implementations from sklearn, RandomizedSearchCV, and GridSearchCV. I tried to use the algorithms from qiskit-machine-learning but they seem not to work as they are not iterable in the same as sklearn ml algorithms are. Is there a plan to build an implementation that works with, e.g. RandomizedSearchCV, in the future? If so, that would be awesome.

Also, the same seems to be true if plugging in the quantumkernels into a basic SVR implementation from sklearn, it's not working with RandomizedSearchCV.

**Comment by javabster** [comment]
@adekusar-drl maybe you have some insight that could be useful here?

**Comment by adekusar-drl** [comment]
It is a well-known issue, unfortunately. Interfaces of the algorithms in qiskit machine learning resemble those from scikit-learn, but the implementations are not fully compatible with scikit-learn. That's why cross validation does not work. And I guess other things from scikit-learn might not work. There's a issue on that https://github.com/Qiskit/qiskit-machine-learning/issues/494. You can take a look at a few possible workarounds mentioned in that issue.
